lib/AbstractHandle/Utils/MongoUtil.py

```diff
@@ -57,14 +57,12 @@ def _get_collection(self, mongo_host, mongo_port, mongo_database, mongo_collecti
 
         return my_collection
 
-    def _inc_counter(self):
-        counter = self.get_hid_counter()
+    def increase_counter(self):
 
-        counter += 1
+        query = {'_id': self.HID_COUNTER_ID}
+        update = {'$inc': {self.HID_COUNTER_ID: 1}}
 
-        update_filter = {'_id': self.HID_COUNTER_ID}
-        update = {'$set': {self.HID_COUNTER_ID: counter}}
-        self.hid_counter_collection.update_one(update_filter, update)
+        return self.hid_counter_collection.find_and_modify(query=query, update=update)['hid_counter']
```

Comment:
Looking at the pymongo docs, this method is deprecated: https://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.find_and_modify

In favor of: https://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.find_one_and_update

Reply:
👍